rspec-core/lib/rspec/core/configuration.rb

```diff
@@ -1765,14 +1766,57 @@ def self.delegate_to_ordering_manager(*methods)
 
       # Set Ruby warnings on or off.
       def warnings=(value)
-        $VERBOSE = !!value
+        @warnings_set = true
+
+        case value
+        when true, false
+          RSpec.deprecate(
+            "RSpec::Core::Configuration#warnings=",
+            :message =>  "config.warnings = #{value} is deprecated, please set warnings to one of: " \
+                         "`:none`, `:all` or `:deprecations_only.`"
+          )
+          value = value ? :all : :none
+        end
+
+        case value
+        when :none
+          $VERBOSE = false
+          self.deprecation_warnings = false
+        when :all
+          $VERBOSE = true
+          self.deprecation_warnings = true
+        when :deprecations_only
+          $VERBOSE = false
+          self.deprecation_warnings = true
+        else
+          raise "Unsupported value for `warnings` (#{value.inspect}). " \
+                "Only `:none`, `:all` and `:deprecations_only` are supported."
+        end
       end
 
       # @return [Boolean] Whether or not ruby warnings are enabled.
       def warnings?
         $VERBOSE
       end
 
+      # @private
+      def warnings_set?
+        @warnings_set
+      end
+
+      if defined?(::Warning) && ::Warning.respond_to?(:[]=)
```

Comment:
Note to self: This is outdated, as we're dropping Ruby 2.x support, and can be simplified.